src/ui/public/visualize/visualize.js

```diff
@@ -24,6 +24,19 @@ define(function (require) {
         esResp: '=?',
       },
       template: require('ui/visualize/visualize.html'),
+      controller: function ($scope) {
+        // Show no results message when hasZeroHits is true and it requires search
+        $scope.showNoResultsMessage = function (hasZeroHits) {
+          var requiresSearch = _.get($scope, 'vis.type.requiresSearch');
+
+          if (requiresSearch && hasZeroHits) {
+            // The metric vis does not show a no results message
+            return _.get($scope, 'vis.type.name') !== 'metric';
```

Comment:
I really don't like looking for `metric` here. Could we add a parameter to the vis type object that says whether or not to pass the data (or lack thereof) through to the visualization to handle?